From 3cd4fefa1d614debf75f059def34ac1aaaf96501 Mon Sep 17 00:00:00 2001
From: Josh Boyer <jwboyer@redhat.com>
Date: Tue, 30 Apr 2013 14:01:57 -0400
Subject: [PATCH] When we encounter a bad PTE, print out what modules were
 loaded, so we can see if there is a common driver potentially at fault.

Signed-off-by: Dave Jones <davej@redhat.com>
---
 mm/memory.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/memory.c b/mm/memory.c
index 6dc1882..99564df 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -57,6 +57,7 @@
 #include <linux/swapops.h>
 #include <linux/elf.h>
 #include <linux/gfp.h>
+#include <linux/module.h>
 #include <linux/migrate.h>
 #include <linux/string.h>
 
@@ -720,6 +721,7 @@ static void print_bad_pte(struct vm_area_struct *vma, unsigned long addr,
 	if (vma->vm_file && vma->vm_file->f_op)
 		printk(KERN_ALERT "vma->vm_file->f_op->mmap: %pSR\n",
 		       vma->vm_file->f_op->mmap);
+	print_modules();
 	dump_stack();
 	add_taint(TAINT_BAD_PAGE, LOCKDEP_NOW_UNRELIABLE);
 }
-- 
1.8.1.4

